api/src/main/java/org/openmrs/api/UserService.java

```diff
@@ -573,4 +573,12 @@
 	 */
 	public User setUserActivationKey(User user);
 	
+	/**
+	 * Change user password given the activation key
+	 * 
+	 * @param activationKey the activation for password reset
+	 * @param newPassword the new password
+	 */
+	public void changeUserPasswordUsingActivationKey(String activationKey, String newPassword) throws APIException;
```

Comment:
APIException is a runtime exception so it's pointless to add it to a throws clause